app/src/common/chromium/com/igalia/wolvic/browser/api/impl/SessionImpl.java

```diff
@@ -138,16 +147,14 @@ public void exitFullScreen() {
     @Override
     public WDisplay acquireDisplay() {
         Log.e("WolvicLifecycle", "acquire display called");
-        WebContents webContents = mRuntime.getContentShellController().createWebContents();
-        BrowserDisplay display = mRuntime.createBrowserDisplay(webContents);
-        WolvicContentRenderView renderView = mRuntime.getRenderView();
-        mRuntime.getContentShellController().getWindowAndroid().setAnimationPlaceholderView(renderView);
-        mDisplay = new DisplayImpl(display, renderView, this);
```

Comment:
This is interesting, why this code has moved to `createBrowserDisplay()`

Reply:
I think it makes sense to not have this in SessionImpl. I would argue the best place to put this is either DisplayImpl or ContentShellController. Howevert, this seems fine for the time being while we are working on the experimental branch. I propose having a separate refactoring before starting to merge this to the main branch to define clear responsibilities for the following classes:
RuntimeImpl, SessionImpl, DisplayImpl, BrowserDisplay and ContentShellController.

right now we're focusing on single tab implementation but when we'll have potentially multiple instances of Tab/WebContents alive it will become more important.

Reply:
I agree with @zakharvoit. I believe this is a temporary place, and if we support the tab feature, the place may change to another place as Zakhar said.